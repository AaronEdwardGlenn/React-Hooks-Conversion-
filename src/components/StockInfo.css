.StockInfo {
    font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin-top: 0px;
    text-align: center;
    padding-top: 20px;

}

.top {
    padding: 20px;
}
.price {
    font-size: 50px;
}



.gif {
    padding-top: 20px;
    height: 20vh;
    width: auto;

}

@media only screen and (max-width: 600px) {
    img {
        width: 90vw;
    }
    .price {
        margin-bottom: 20px;
    }
}
