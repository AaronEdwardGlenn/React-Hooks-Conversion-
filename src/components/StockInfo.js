import React from 'react';
import PropTypes from 'prop-types';
import styles from './StockInfo.css';


const StockInfo = ({ symbol, price, dailyChange }) => {
  let change = dailyChange; 
  let arr = [];

  if(change) {
    arr.push(change.split(''));
  }
  let dailyChangeSplit = arr[0];
  let arr2 = []; 

  if(dailyChangeSplit) {
    arr2.push(dailyChangeSplit[0]);
  }

  let dailyChangeStyle;

  if(arr2[0] === '-'){
    dailyChangeStyle = { color: 'red' };
  } else {
    dailyChangeStyle = { color: 'green' };
  }

  return (
    <>
      <div className={styles.StockInfo}>
        <p> Symbol: {symbol} </p> 
        <p>  Price: {price} </p>
        <p style={dailyChangeStyle}>  <span style={{ color: 'black' }}>Daily Change: </span> {dailyChange} </p>
      </div>
    </>
  );};

StockInfo.propTypes = {
  symbol: PropTypes.string.isRequired,
  price: PropTypes.string,
  dailyChange: PropTypes.string,
};

export default StockInfo;
