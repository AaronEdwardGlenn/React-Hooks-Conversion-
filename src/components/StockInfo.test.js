import React from 'react';
import { shallow } from 'enzyme';
import StockInfo from './StockInfo'
;

describe('the stockinfo component', () => {
  it('can match the snap', () => {
    const wrapper = shallow(<StockInfo price="420" dailyChange="Calvin Coolidge" symbol="YEE" />);
    expect(wrapper).toMatchSnapshot(); 
  });
    
});
