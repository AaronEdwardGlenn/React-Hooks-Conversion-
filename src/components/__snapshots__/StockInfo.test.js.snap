// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the stockinfo component can match the snap 1`] = `
<div>
  Symbol:
  YEE
  Price: 
  420
  Daily Change: 
  Calvin Coolidge
</div>
`;
